scripts/deploy/pre-updb

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+IFS=$'\n\t'
+set -euo pipefail
+
+#
+# Pre-deploy updb
+#
+# This will perform an drush updb on an environment when
+# the environment has the GOVCMS_DEPLOY_PRE_UPDB variable defined.
+#
+# In some use cases databaes updates need to be performed early in
+# the deploy process, even before the site has been bootstrapped
+# (for example with drush status). This provides the deploy agent
+# the ability to perform updates early in the automated flow.
+#
+# The ansible deploy process will add GOVCMS_DEPLOY_PRE_UPDB prior to
+# triggering the deploy to allow this to run properly.
+#
+
+LAGOON_ENVIRONMENT_TYPE=${LAGOON_ENVIRONMENT_TYPE:-production}
+GOVCMS_DEPLOY_PRE_UPDB=${GOVCMS_DEPLOY_PRE_UPDB:-false}
+
+if [ "$GOVCMS_DEPLOY_PRE_UPDB" = false ]; then
+  # A pre-deploy update is not required so return early.
+  exit 0;
+fi
+
+echo "GovCMS Deploy :: Pre-deploy database updates"
+echo "Environment type: $LAGOON_ENVIRONMENT_TYPE"
+
+drush updatedb -y
```

Comment:
is there scope to broaden this to insert a command in this step from an env_var? - in most cases it may be an updb, but it could also equally be a drush pmu/en or a drush cset?

Reply:
I think I like the flexibility of that, I'm not sure that being able to execute arbitrary commands fits the atomic pattern here. I've added a new `pre-deploy-commands` script which we should probably discuss before including, but basically what this will do is.

- Look for an env var defined as `GOVCMS_DEPLOY_PRE_COMMANDS`
- If that is not defined it'll use `$1`
- If neither are defined it will skip
- It will match the command given on a whitelist of drush commands that we will allow for example it's currently matching on `^(pmu|pm:uninstall|config:set|cset).*`

Reply:
nice!